File: src/Test/LazyEvaluationFFICheatcode.erl
-module(test_lazyEvaluationFFICheatcode@foreign).
-export([runLazyEvaluationFFICheatcode/1]).
runLazyEvaluationFFICheatcode(N) -> sum(N, 0).
sum(0, Acc) -> Acc;
sum(N, Acc) -> sum(N - 1, Acc + N).

File: src/Test/LazyEvaluationFFICheatcode.php
<?php
$exports['runLazyEvaluationFFICheatcode'] = function($limit) {
    $n = (int)$limit;
    $acc = 0;
    for ($i = 0; $i < $n; $i++) {
        $acc += 1000;
    }
    return $acc;
};
return $exports;
